pkg/connpool/dialer.go

```diff
@@ -117,7 +117,7 @@ func (h *dialer) handleControl(ctx context.Context, cm Control) {
 	case ConnectOK:
 		go h.writeLoop(ctx)
 		go h.readLoop(ctx)
-	case Disconnect:
+	case Disconnect, ConnectReject:
```

Comment:
I think this is more similar to a `WriteClosed` in that there's no point in sending a `DisconnectOK` back.

Reply:
Yeah that makes sense... and "Disconnect" implies a connection, which we don't have if we get "ConnectReject" so I have made the chnage!